src/System-SessionManager/SessionAccessModeResolver.class.st

```diff
@@ -0,0 +1,49 @@
+"
+I check the image arguments and resolve if the access mode of the image. I am started very soon in the image start-up process.
+"
+Class {
+	#name : #SessionAccessModeResolver,
+	#superclass : #Object,
+	#category : #'System-SessionManager-Utilities'
+}
+
+{ #category : #'class initialization' }
+SessionAccessModeResolver class >> initialize [
+
+	"self initialize"
+
+	SessionManager default
+		register: (ClassSessionHandler forClassNamed: self name)
+		inCategory: SessionManager default systemCategory  
+		atPriority: SessionManager default defaultPriority - 90.
+]
+
+{ #category : #'class initialization' }
+SessionAccessModeResolver class >> resolve [ 
+
+	"check if the image was started in the read-only mode"
+
+	| arguments mode | 
+	
+	arguments := Smalltalk argumentsStartingAtIndex: 0.
```

Comment:
I am not against your approach.
But I assumed it will be something simpler directly in session creation method:
```Smalltalk
 SessionManager>>newSession
	| aWorkingSession |
	aWorkingSession := WorkingSession new.
	aWorkingSession manager: self.
	aWorkingSession readPropertiesFrom: (self environment argumentsStartingAtIndex: 0).
	^ aWorkingSession
```

Reply:
I think that as part of the startup list it will be more clear and more fitting into the current startup architecture and with a standalone class, we can extend the responsibility of it. In the end, the access mode may be a standalone class(es)

Reply:
Ok. 
But my idea was also introduce nice API to query startup options.
With this approach it still not exists.
So the possible "next StartupOptionResolver" will again do the anySatisfy loops with arguments.